src/index_runner/es_indexers/main.py

```diff
@@ -32,52 +30,74 @@ def index_obj(obj_data, ws_info, msg_data):
         msg_data - json event data received from the kafka workspace events
             stream. Must have keys for `wsid` and `objid`
     """
-    obj_type = obj_data['info'][2]
+    obj_type = obj_info[2]
     (type_module, type_name, type_version) = ws_utils.get_type_pieces(obj_type)
     if (type_module + '.' + type_name) in _TYPE_BLACKLIST:
         # Blacklisted type, so we don't index it
+        logger.debug(f'Skipping blacklisted type {type_module}, {type_name}')
         return
+
     # check if this particular object has the tag "noindex"
-    metadata = ws_info[-1]
-    # If the workspace's object metadata contains a "nosearch" tag, skip it
-    if metadata.get('searchtags'):
-        if 'noindex' in metadata['searchtags']:
+    ws_metadata = ws_info[-1]
+    # If the workspace's object metadata contains a "noindex" tag, skip it
+    if ws_metadata.get('searchtags'):
+        if 'noindex' in ws_metadata['searchtags']:
+            logger.debug(f'Skipping "noindex" workspace object {ws_info[0]}, {obj_info[6]}')
             return
+
+    # We may refuse to index very large objects
+    # TODO: should be very high in prod, lower for development
+    max_object_size = config().get('max_object_size')
```

Comment:
I think we should construct the indexes to accommodate for this.  I can't think of an instance where we would want this to happen.